app/views/ShareView/index.js

```diff
@@ -1,307 +1,360 @@
-								testID='send-message-share-view'
-								buttonStyle={styles.send}
-							/>
-						</CustomHeaderButtons>
-					)
-					: null)
-		});
+		const {
+			room, thread, readOnly, attachments
+		} = this.state;
+		const { navigation, theme } = this.props;
+
+		const options = {
+			headerTitle: () => <Header room={room} thread={thread} />,
+			headerTitleAlign: 'left',
+			headerTintColor: themes[theme].previewTintColor
+		};
+
+		// if is share extension show default back button
+		if (!this.isShareExtension) {
+			options.headerLeft = () => <CloseModalButton navigation={navigation} buttonStyle={{ color: themes[theme].previewTintColor }} />;
+		}
+
+		if (!attachments.length && !readOnly) {
+			options.headerRight = () => (
+				<CustomHeaderButtons>
+					<Item
+						title={I18n.t('Send')}
+						onPress={this.send}
+						buttonStyle={[styles.send, { color: themes[theme].previewTintColor }]}
+					/>
+				</CustomHeaderButtons>
+			);
+		}
+
+		options.headerBackground = () => <View style={[styles.container, { backgroundColor: themes[theme].previewBackground }]} />;
+
+		navigation.setOptions(options);
 	}
 
-	setReadOnly = async() => {
+	// fetch server info
+	getServerInfo = async() => {
+		const { server } = this.props;
+		const serversDB = database.servers;
+		const serversCollection = serversDB.collections.get('servers');
+		try {
+			this.serverInfo = await serversCollection.find(server);
+		} catch (error) {
+			// Do nothing
+		}
+	}
+
+	getReadOnly = async() => {
 		const { room } = this.state;
 		const { user } = this.props;
-		const { username } = user;
-		const readOnly = await isReadOnly(room, { username });
-
-		this.setState({ readOnly, canSend: !(readOnly || isBlocked(room)) }, () => this.setHeader());
+		const readOnly = await isReadOnly(room, user);
+		return readOnly;
 	}
 
-	bytesToSize = bytes => `${ (bytes / 1048576).toFixed(2) }MB`;
+	getAttachments = async() => {
+		const { mediaAllowList, maxFileSize } = this.state;
+		const items = await Promise.all(this.files.map(async(item) => {
+			// Check server settings
+			const { success: canUpload, error } = canUploadFile(item, mediaAllowList, maxFileSize);
+			item.canUpload = canUpload;
+			item.error = error;
+
+			// get video thumbnails
+			if (item.mime?.match(/video/)) {
+				try {
+					const { uri } = await VideoThumbnails.getThumbnailAsync(item.path);
+					item.uri = uri;
+				} catch {
+					// Do nothing
+				}
+			}
+
+			// Set a filename, if there isn't any
+			if (!item.filename) {
+				item.filename = new Date().toISOString();
+			}
+			return item;
+		}));
+		return {
+			attachments: items,
+			selected: items[0]
+		};
+	}
 
-	sendMessage = async() => {
-		const { isMedia, loading } = this.state;
+	send = async() => {
+		const { loading, selected } = this.state;
 		if (loading) {
 			return;
 		}
 
-		this.setState({ loading: true });
-		if (isMedia) {
-			await this.sendMediaMessage();
+		// update state
+		await this.selectFile(selected);
```

Comment:
This should be done only when it's a media share.

Reply:
`selectFile` doesn't return a promise, so we can't wait for this.
maybe it can be fixed solving a promise at the `this.setState` callback function.